src/thunderbots/software/ai/navigator/obstacle.h

```diff
@@ -0,0 +1,20 @@
+// An obstacle is an area to avoid based on the size of the robot and its velocity
+#pragma once
+
+#include "geom/polygon.h"
+#include "geom/point.h"
+#include "geom/util.h"
+#include "util/parameter/dynamic_parameters.h"
+
+class Obstacle
+{
+   public:
+    /*
+     * Gets the buffer zone around the given robot obstacle that other robots
+     * should not enter
+     * @param robot robot to create obstacle buffer zone around
+     * @param avoid_factor safety factor to vary the size of the buffer zone (1=default)
```

Comment:
This `avoid_factor` is a multiplier? As in it scales some default value? I realize this is a bit harder to define since the boundary polygon also accounts for velocity, but IMO "scaling" values like this are a bit tough for the caller to work with because they don't know what they're scaling, so it's hard pick an appropriate value.

Maybe we could use 2 values, where 2 is the "radius" to avoid the robot by, and the other is a scaling factor for how far the polygon extends based on velocity.

@jordanzliu 	thoughts?

Reply:
The real question is what should be a dynamic params and what should be passed in. The velocity scaling is covered by dynamic params atm.

Reply:
I would suggest something like: "The buffer radius around the robot in meters. This will "expand" the robot radius by the given value before converting it to a polygon".

I'm not super sure where this velocity scaling is? In general though, we should try to avoid "gofer holes" where dynamic parameters just pop up in the middle of functions or classes. This is for a few reasons:
- it makes things harder to test, what if the dynamic parameter changes and causes whatever we're testing to fail?
- it increases coupling, now the function or class is dependent on a component that is unrelated to it

So i'd say the rule is pretty much, if you're unit testing it, probably don't want dynamic parameters popping up in it, we should prefer to pass them in.

Reply:
I added those parameters and placeholder. See https://github.com/UBC-Thunderbots/Software/pull/501